hilti/toolchain/src/compiler/cxx/elements.cc

```diff
@@ -669,10 +700,10 @@ cxx::Formatter& cxx::operator<<(cxx::Formatter& f, const cxx::Block& x) {
 cxx::Formatter& cxx::operator<<(cxx::Formatter& f, const cxx::Expression& x) { return f << std::string(x); }
 
 cxx::Formatter& cxx::operator<<(cxx::Formatter& f, const cxx::ID& x) {
-    if ( x.namespace_() == f.namespace_() )
+    if ( f.namespace_() && x.namespace_() == cxx::ID(*f.namespace_()) )
```

Comment:
If we made `Formatter::namespace_()` return a `cxx::ID` this would simply be

```suggestion
    if ( f.namespace_() == x.namespace_() )
```